counsel.el

```diff
@@ -191,11 +191,9 @@ Update the minibuffer with the amount of lines collected every
       (with-current-buffer (process-buffer process)
         (goto-char (point-min))
         (setq size (- (buffer-size) (forward-line (buffer-size))))
-        (ivy--set-candidates
-         (split-string
-          (buffer-string)
-          counsel-async-split-string-re
-          t)))
+        (let ((cands (split-string (buffer-string) counsel-async-split-string-re t)))
+          (setq ivy--old-cands ivy--all-candidates)
```

Comment:
Sorry, the purpose of the variables wasn't entirely clear to me.  Thanks for the explanation!

I think I added that because `ivy--old-cands` is being updated in `counsel--async-sentinel` at the end of the async process (for either of the two events its matching on), but not during the consumption of the output in `counsel--async-filter`.

What I was seeing was that If the process was killed by running `C-c C-o` before the process had completed (the event would be "killed: [0-9]+\n" and is more obvious with ack because its slower than ag), I wasn't winding up with the expected results in the occur buffer.  A "killed [0-9]+\n" event isn't handled in `counsel--async-sentinel`.  Perhaps it should be?

But, Its also possible it was something I added while trying to debug that issue and it isn't actually needed.

I'll take another look.